keyboards/mxss/mxss.c

```diff
@@ -0,0 +1,199 @@
+/* Copyright 2018 Jumail Mundekkat / MxBlue
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#include QMK_KEYBOARD_H
+#include "tmk_core/common/eeprom.h"
+#include "tmk_core/common/action_layer.h"
+#include "rgblight.h"
+#include "mxss_frontled.h"
+
+// Variables for controlling front LED application
+uint8_t fled_mode;  // Mode for front LEDs
+uint8_t fled_val;   // Brightness for front leds (0 - 255)
+LED_TYPE fleds[2];  // Front LED rgb values for indicator mode use
+
+// External values from keymap.c for front LED control
+extern hs_set layer_colors[];
+extern const size_t lc_size;
+
+void matrix_init_kb(void) {
+    // If EEPROM config exists, load it
+    if (eeprom_is_valid()) {
+        fled_config fled_conf;
+        fled_conf.raw = eeprom_read_byte(EEPROM_FRONTLED_ADDR);
+        fled_mode = fled_conf.mode;
+        fled_val = fled_conf.val * FLED_VAL_STEP;
+    // Else, default config
+    } else {
+        fled_mode = FLED_RGB;
+        fled_val = 10 * FLED_VAL_STEP;
+        eeprom_update_conf();   // Store default config to EEPROM
+    }
+    
+    // Set default values for leds
+    setrgb(0, 0, 0, &fleds[0]);
```

Comment:
If you've tried this code, I'm guessing that you've had some issues on startup? 

If so... that's because `rgblight_init` is called AFTER `matrix_init_kb`, resetting the value. 

You'd be better off calling this in the first matrix scan, as that should be the right order.

Reply:
There are no issues, as it sets some local LEDs that are copied on rgblight_set() call.

Reply:
Ah, okay.  I know that it's an issue for the official rgb code, so wanted to make sure.